.github/workflows/manual-publish-alarm-to-dev.yaml

```diff
@@ -0,0 +1,58 @@
+---
+name: "Publish Alarm SAM Template to Dev Env"
+
+on:
+  push:
+    paths:
+      - "src/infra/alarm/**"
+      - ".github/workflows/manual-publish-alarm-to-dev.yaml"
+    branches:
+      - main
+  workflow_dispatch:
+    inputs:
+      gitRef:
+        description: "Input branch name, commit SHA, or tag"
+        required: true
+        type: string
+        default: main
+
+permissions:
+  id-token: write
+  contents: read
+
+jobs:
+  validate_deployment:
+    name: Deployment Template Checks
+    uses: ./.github/workflows/validate-alarm-sam-template.yaml
+    with:
+      gitRef: ${{ inputs.gitRef }}
+    secrets:
+      githubActionsRoleARN: ${{ secrets.DEV_ALARM_GH_ACTIONS_ROLE_ARN }}
+
+  publish_artifacts:
+    name: "Publish Template"
+    runs-on: ubuntu-latest
+    timeout-minutes: 60
+    needs:
+      - validate_deployment
+    steps:
+      - uses: actions/checkout@83b7061638ee4956cf7545a6f7efe594e5ad0247 # pin@v3
+        with:
+          ref: ${{ inputs.gitRef }}
+
+      - name: Configure AWS Credentials
+        uses: aws-actions/configure-aws-credentials@e1e17a757e536f70e52b5a12b2e8d1d1c60e04ef # pin@v1-node16
+        with:
+          role-to-assume: ${{ secrets.DEV_ALARM_GH_ACTIONS_ROLE_ARN }}
+          aws-region: eu-west-2
+
+      - name: "Deploy SAM app"
+        uses: alphagov/di-devplatform-upload-action@1db904c002e0f98dbcb5ab453ceb1b71de5f7dd3 # pin@v2
```

Comment:
could we try using the govuk-one-login actions. 

https://github.com/govuk-one-login/devplatform-upload-action
https://github.com/govuk-one-login/devplatform-upload-action-ecr

eg `govuk-one-login/devplatform-upload-action-ecr` note they have dropped the `di-`

Reply:
will also need to get the commit sha

Reply:
Shall we do this across VC Storage and AIS workflows in scope of a separate ticket?

Reply:
@TaherKapasi